src/endpoint.rs

```diff
@@ -476,8 +478,8 @@ impl Endpoint {
     }
 
     /// Get an existing connection for the peer address.
-    pub(crate) async fn get_connection(&self, peer_addr: &SocketAddr) -> Option<Connection> {
-        if let Some((conn, guard)) = self.connection_pool.get(peer_addr).await {
+    pub(crate) async fn get_connection(&self, peer_addr: &SocketAddr) -> Option<Connection<I>> {
```

Comment:
we probably need `get_connection_by_id` too no? 

In Safe we'd want to avoid any referencing via `SockerAddr`